crates/pgt_workspace/src/features/completions.rs

```diff
@@ -24,3 +28,167 @@ impl IntoIterator for CompletionsResult {
         self.items.into_iter()
     }
 }
+
+pub(crate) fn get_statement_for_completions<'a>(
```

Comment:
why can't this be implemented as a `Filter` in `parsed_document`?

Reply:
because I need to peek into the next statement for the overlap check :/

Reply:
makes sense. maybe we can find a good abstraction for this but for now lets roll with this.